src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Commands.Package.Update
+{
+    internal static class PackageUpdateCommandRunner
+    {
+        internal static async Task<int> Run(PackageUpdateArgs args, ILoggerWithColor logger, IDGSpecFactory dGSpecFactory, MSBuildAPIUtility msbuild, CancellationToken cancellationToken)
+        {
+            // 1. Get DGSpec for project/solution
+            // 2. Find suitable version of package(s) to update
+            // 3. Preview restore to validate changes
+            // 4. Update MSBuild files
+            // 5. Commit restore if everything successful
+
+            await Task.Delay(1, cancellationToken);
```

Comment:
What is the purpose of the 1ms delay here? If its for cancellation, this is an odd pattern.

Reply:
since we raise warnings to errors, it was breaking when the method signature said async, but I didn't yet have an await in the method body.  I just forgot to remove it as I'm rushing too much. Usuually I carefully self-review my own PRs, but this time I didn't 😞